code/src/sixsq/nuvla/server/resources/spec/voucher.cljc

```diff
@@ -0,0 +1,189 @@
+(ns sixsq.nuvla.server.resources.spec.voucher
+  (:require
+    [clojure.spec.alpha :as s]
+    [sixsq.nuvla.server.resources.spec.common :as cimi-common]
+    [sixsq.nuvla.server.resources.spec.core :as cimi-core]
+    [sixsq.nuvla.server.util.spec :as su]
+    [spec-tools.core :as st]))
+
+
+(s/def ::owner ::cimi-common/resource-link)
+(s/def ::user ::cimi-common/resource-link)
+
+(s/def ::amount
+  (-> (st/spec double)
+      (assoc :name "amount"
+             :json-schema/type "double"
+             :json-schema/editable true
+
+             :json-schema/display-name "voucher amount"
+             :json-schema/description "amount of the voucher"
+             :json-schema/group "body"
+             :json-schema/order 20
+             :json-schema/hidden false
+             :json-schema/sensitive false)))
+
+
+(s/def ::currency
+  (-> (st/spec ::cimi-core/nonblank-string)
```

Comment:
It would be much better to limit this value to the ones defined in ISO 4217.  Using a regular expression limiting to 3 uppercase characters, along with a reference to ISO 4217, would be enough.

Reply:
ok, for which codes? CHF EUR USD?

Reply:
done with a defined list